src/util/TypeTraits.h

```diff
@@ -118,25 +128,47 @@ concept SimilarTo = isSimilar<T, U>;
 
 /// True iff `T` is similar (see above) to any of the `Ts...`.
 template <typename T, typename... Ts>
-concept isTypeAnyOf = (... || isSimilar<T, Ts>);
+concept SimilarToAny = (... || isSimilar<T, Ts>);
 
-/// isTypeContainedIn<T, U> It is true iff type U is a pair, tuple or variant
-/// and T `isSimilar` (see above) to one of the types contained in the tuple,
-/// pair or variant.
+/// True iff `T` is the same as any of the `Ts...`.
 template <typename T, typename... Ts>
-constexpr static bool isTypeContainedIn = false;
+concept SameAsAny = (... || std::same_as<T, Ts>);
 
-template <typename T, typename... Ts>
-constexpr static bool isTypeContainedIn<T, std::tuple<Ts...>> =
-    isTypeAnyOf<T, Ts...>;
+/*
+The implementation for `SimilarToAnyTypeIn` and `SameAsAnyTypeIn` (see below
+namespace detail).
+*/
+namespace detail {
+template <typename T, typename Template>
+struct SimilarToAnyTypeInImpl;
 
-template <typename T, typename... Ts>
-constexpr static bool isTypeContainedIn<T, std::variant<Ts...>> =
-    isTypeAnyOf<T, Ts...>;
+template <typename T, template <typename...> typename Template, typename... Ts>
+struct SimilarToAnyTypeInImpl<T, Template<Ts...>>
+    : std::integral_constant<bool, SimilarToAny<T, Ts...>> {};
 
-template <typename T, typename... Ts>
-constexpr static bool isTypeContainedIn<T, std::pair<Ts...>> =
-    isTypeAnyOf<T, Ts...>;
+template <typename T, typename Template>
+struct SameAsAnyTypeInImpl;
+
+template <typename T, template <typename...> typename Template, typename... Ts>
+struct SameAsAnyTypeInImpl<T, Template<Ts...>>
+    : std::integral_constant<bool, SameAsAny<T, Ts...>> {};
+
+}  // namespace detail
+/*
+SimilarToAnyTypeIn<T, U> It is true iff type U is a pair, tuple or variant
+and T `isSimilar` (see above) to one of the types contained in the tuple,
+pair or variant.
+*/
+template <typename T, typename Template>
+concept SimilarToAnyTypeIn = detail::SimilarToAnyTypeInImpl<T, Template>::value;
+
+/*
+SameAsAnyTypeIn<T, U> It is true iff type U is a pair, tuple or variant
+and T `std::same_as` to one of the types contained in the tuple,
+pair or variant.
+*/
```

Comment:
Simply write "Equivalent to  `SimilarToAnyTypeIn` (see above), but checks for exactly matching types via `same_as`.

Reply:
Done, see [b751bae](https://github.com/ad-freiburg/qlever/pull/1214/commits/b751baeed2903bf1699a55f8b333d2df30e4dfbf).